crates/sui-types/src/temporary_store.rs

```diff
@@ -565,31 +580,39 @@ impl<S> TemporaryStore<S> {
         Ok(())
     }
 
+    /// TODO: can we make this more efficient e.g by minimizing clones
     pub fn to_effects(
-        mut self,
+        &mut self,
         shared_object_refs: Vec<ObjectRef>,
         transaction_digest: &TransactionDigest,
         transaction_dependencies: Vec<TransactionDigest>,
         gas_cost_summary: GasCostSummary,
         status: ExecutionStatus,
         gas: &[ObjectRef],
         epoch: EpochId,
-    ) -> (InnerTemporaryStore, TransactionEffects) {
+        check_effects_size: bool,
```

Comment:
nit suggestion - maybe "enforce_" would be a bit clearer that it's actually failing in case of too large rather than just checking it to log an error or something?